### PR TITLE
libswoc - replace string_view_util with libswoc version.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
This is needed both to remove redundancy and to avoid circular dependencies, as libswoc internals depend on the functions in the `string_view` utils.

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
[approve ci fedora]

File: lib/swoc/include/swoc/string_view_util.h [hunk 1]
```diff
@@ -85,15 +74,14 @@ strcmp(const std::string_view &lhs, const std::string_view &rhs)
  * @code
  *   memcpy(dst, src.data(), size.size());
  * @endcode
- * Therefore @a dst must point at a buffer large enough to hold @a src. If this is not already
+ * Therefore @a dst must point at a buffer large enought to hold @a src. If this is not already
```
Comment (bneradt) [inline lib/swoc/include/swoc/string_view_util.h:77]: enought

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks good.

**Comment by bneradt** [comment]
The builds are failing due to linking issues. Here's a link to the Ubuntu failure:
https://ci.trafficserver.apache.org/job/Github_Builds/job/ubuntu/389/console

**Comment by SolidWallOfCode** [comment]
Oh, I'm tracking the linking issues. The problem is all of that works perfectly locally. This is part of resolving the linking issues with #9193 .

**Comment by SolidWallOfCode** [comment]
[approve ci rocky]

**Comment by SolidWallOfCode** [comment]
Seriously? That was the fix? Gah.

**Comment by SolidWallOfCode** [comment]
Seriously? That fix worked? Gaaaaaaaaaaaaaaaaaaaaaaah!

**Review by bneradt — APPROVED** [review]

**Pull request merged**